cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+package io.aiven.klaw.clusterapi.services;
+
+import io.aiven.klaw.clusterapi.models.AclIPPrincipleType;
+import io.aiven.klaw.clusterapi.models.AclPatternType;
+import io.aiven.klaw.clusterapi.models.ApiResultStatus;
+import io.aiven.klaw.clusterapi.models.ClusterAclRequest;
+import io.aiven.klaw.clusterapi.models.KafkaSupportedProtocol;
+import io.aiven.klaw.clusterapi.models.RequestOperationType;
+import io.aiven.klaw.clusterapi.utils.ClusterApiUtils;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeAclsResult;
+import org.apache.kafka.common.acl.AccessControlEntry;
+import org.apache.kafka.common.acl.AccessControlEntryFilter;
+import org.apache.kafka.common.acl.AclBinding;
+import org.apache.kafka.common.acl.AclBindingFilter;
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.acl.AclPermissionType;
+import org.apache.kafka.common.resource.PatternType;
+import org.apache.kafka.common.resource.ResourcePattern;
+import org.apache.kafka.common.resource.ResourcePatternFilter;
+import org.apache.kafka.common.resource.ResourceType;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class ApacheKafkaAclService {
+
+  private static final long TIME_OUT_SECS_FOR_ACLS = 5;
+
+  private Environment env;
+
+  private ClusterApiUtils clusterApiUtils;
+
+  public ApacheKafkaAclService(ClusterApiUtils clusterApiUtils) {
+    this.clusterApiUtils = clusterApiUtils;
+  }
+
+  public ApacheKafkaAclService() {}
+
+  public ApacheKafkaAclService(Environment env) {
+    this.env = env;
+  }
+
+  public synchronized Set<Map<String, String>> loadAcls(
+      String environment, KafkaSupportedProtocol protocol, String clusterName) throws Exception {
+    log.info("loadAcls {} {}", environment, protocol);
+    Set<Map<String, String>> acls = new HashSet<>();
+
+    AdminClient client = clusterApiUtils.getAdminClient(environment, protocol, clusterName);
+    if (client == null) {
+      throw new Exception("Cannot connect to cluster.");
+    }
+
+    try {
+      AclBindingFilter aclBindingFilter = AclBindingFilter.ANY;
+      DescribeAclsResult aclsResult = client.describeAcls(aclBindingFilter);
+
+      aclsResult
+          .values()
+          .get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS)
+          .forEach(
+              aclBinding -> {
+                filterAndUpdateAclBindings(acls, aclBinding);
+              });
+    } catch (Exception e) {
+      log.error("Error " + e.getMessage());
```

Comment:
```suggestion
      log.error("Exception: ", e);
```
it's better to log the whole trace

Reply:
This is still not fixed, so put it back to unresolved